@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -513,6 +517,13 @@ export class TileGeometryLoader {
         return true;
     }
 
+    /**
+     * `True` if TileGeometryLoader was canceled
+     */
+    private get isCanceled(): boolean {
+        return this.m_state === TileGeometryLoaderState.Canceled;
+    }
+
```

Comment:
Can you have a look at TileGeometryLoader.discardNeedlessTile, it has not much to do with the tileGEometryLoader, and has some back reference to the visibleTileSet for disposal, keeping this in tile, might make things easier, maybe it could become something like  if(!tile.needed){tile.dispose()}

Reply:
I did something in the middle, let me know what you think.

Reply:
I left a comment there